bundles/org.openhab.core.persistence/src/main/java/org/openhab/core/persistence/extensions/PersistenceExtensions.java

```diff
@@ -1217,20 +1334,21 @@ private static void internalPersist(Item item, TimeSeries timeSeries, @Nullable
      * @param item the {@link Item} to get the variance for
      * @param begin the point in time from which to compute
      * @param end the end time for the computation
+     * @param type LEFT, RIGHT, MIDPOINT or TRAPEZOIDAL representing approximation types for Riemann sums
      * @param serviceId the name of the {@link PersistenceService} to use
      * @return the variance between both points of time, or <code>null</code> if <code>begin</code> is after
      *         <code>end</code>, if the persistence service given by
      *         <code>serviceId</code> is not available, or it is not a {@link QueryablePersistenceService}, or it is not
      *         a {@link QueryablePersistenceService}, or if there is no persisted state for the
      *         given <code>item</code> between <code>begin</code> and <code>end</code>
      */
-    public static @Nullable State varianceBetween(Item item, ZonedDateTime begin, ZonedDateTime end,
+    public static @Nullable State varianceBetween(Item item, ZonedDateTime begin, ZonedDateTime end, RiemannType type,
             @Nullable String serviceId) {
-        return internalVarianceBetween(item, begin, end, serviceId);
+        return internalVarianceBetween(item, begin, end, type, serviceId);
     }
 
     private static @Nullable State internalVarianceBetween(Item item, @Nullable ZonedDateTime begin,
-            @Nullable ZonedDateTime end, @Nullable String serviceId) {
+            @Nullable ZonedDateTime end, RiemannType type, @Nullable String serviceId) {
```

Comment:
This comment applies to all the other similar methods.

Whilst implementing this into the jruby helper library, I came across one particular issue.

Could you please make the `type` parameter `@Nullable`, and provide the default (i.e. `RiemannType.LEFT`) inside the `internalXxxxXxxx` methods, and pass `null` by the callers? This would:
- DRY and collect all the defaults into one place (in the internalXxxx method) rather than in the callers
- Makes it easy for the helper libraries to also call it with `null` to indicate "default", and therefore letting core decides what the default is.

As it is now, I'd either have to also hard-code this default in the helper library, thus maintaining a duplicate default, or conditionally call one of the overloaded methods that doesn't take the `type` argument in order to use the "default" type.

Reply:
Yes, I will do that. Makes sense.

Reply:
Done